.travis.yml

```diff
@@ -44,6 +44,22 @@ matrix:
       jdk: oraclejdk8
       after_success: bash <(curl -s https://codecov.io/bash) -f target/coverage/codecov.json
 
+  exclude:
+    # CLJS doesn't build on JDK9 because it depends on a package (javax.xml.bind)
+    # deprecated in JDK9. See (CLJS-2377).
+    # After a CLJS release, the VERSION=1.9 build can be re-enabled.
```

Comment:
In https://github.com/bhauman/lein-figwheel/issues/612 another way to handle this was discussed.  Conditionally adding ` :jvm-opts ["--add-modules" "java.xml.bind"]` to the build.

Reply:
I considered doing this. So we could add something like:
```
:jvm-opts ~(try (do (Class/forName "javax.xml.bind.DatatypeConverter")
                     [])
                (catch ClassNotFoundException e  ["--add-modules" "java.xml.bind"]))
```

But this seemed like too much of a hack just to get the tests working. Should we  take this path ?

Reply:
On second thought, I should've mentioned this in the PR as well. Thanks for bringing it up.

Reply:
No worries, I just wanted to put it out there. I don't have a strong opinion either way.

However, that said, another equally hacky approach would be to detect the java version in the Makefile and use `lein update-in` to inject the `:jvm-opts` as needed for cljs.

The Makefile has been the place where other transient toggling is being done, for instance filtering out some cloverage tests, and turning off mranderson.

Reply:
Ah, that's true. One advantage of this would be that all toggling would be in one place.

But would doing this in Clojure be a bit safer than grepping the version there ? (I have to admit that I have minimal experience with Makefiles, so I might be totally wrong. I used `Class/forName ` because of Alex Miller's  [comment](https://github.com/bhauman/lein-figwheel/issues/612#issuecomment-353169205) on the same figwheel issue.)

If we're not disabling the tests and adding java-opts instead, I'm leaning toward adding it in the `project.clj` file and using `Class/forName`. But I'm fine with the Makefile version too.

Reply:
Well, for instance in orchard there's this
https://github.com/clojure-emacs/orchard/blob/master/Makefile#L5-L8

which detects Java 9 and constructs a test selector `:java9.0` which is then used to filter certain test cases based on metadata (`lein with-profile <+1.9,...etc.> test :java9.0`).  The usage would be different in this case, but we're already taking a similar risk elsewhere.